internal/controllers/addon/addon_deletion_reconciler.go

```diff
@@ -0,0 +1,118 @@
+package addon
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+)
+
+const (
+	defaultDeleteTimeoutDuration = time.Hour * 1
+	DELETION_RECONCILER_NAME     = "deletionReconciler"
+)
+
+type addonDeletionStrategy interface {
+	NotifyAddon(context.Context, *addonsv1alpha1.Addon) error
+	AckReceivedFromAddon(context.Context, *addonsv1alpha1.Addon) (bool, error)
+}
+
+type clock interface {
+	Now() time.Time
+}
+
+type defaultClock struct{}
+
+func (c defaultClock) Now() time.Time {
+	return time.Now()
+}
+
+type addonDeletionReconciler struct {
+	clock      clock
+	strategies []addonDeletionStrategy
```

Comment:
```suggestion
	handlers []addonDeletionHandler
```

as we discussed, let us use a name like `handlers` as this a `1 or more` association instead of an `either or` association.

Reply:
👍